### PR TITLE
{tools}[intel/2018a] hypothesis v3.66.18 w/ Python 3.6.4

### PR DESCRIPTION
**Pull request by SimonPinches**
(created using `eb --new-pr`)

### CODE REVIEW

**Review by smoors — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/h/hypothesis/hypothesis-3.66.18-intel-2018a-Python-3.6.4.eb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    ('coverage', '5.3', {
+        'source_urls': ['https://pypi.python.org/packages/source/c/coverage'],
+        'checksums': ['280baa8ec489c4f542f8940f9c4c2181f0306a8ee1a54eceba071a449fb870a0'],
+    }),
+    (name, version, {
+        'source_urls': ['https://pypi.python.org/packages/source/h/hypothesis/'],
+        'checksums': ['6bcf0e565cd394080d692f4595215335a90a868325aa4869dfe7450be9a8a583'],
+    }),
+]
+
```
[the remark this answers is not in the document]
Comment (SimonPinches) [reply easybuild/easyconfigs/h/hypothesis/hypothesis-3.66.18-intel-2018a-Python-3.6.4.eb:36]: Done as suggested @smoors.

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node368.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/7ee9d20593c0dc7ec38c5427c54dc96d for a full test report.

**Comment by smoors** [comment]
@SimonPinches `pip check` claims there is an extension missing:

> hypothesis 3.66.18 requires enum34, which is not installed.

which makes no sense since Python3.6 already has enum in the standard lib.

the culprit however is setuptools, which claims to be version 28.8.0 (at least in our site), which again makes no sense since version 38.4.0 is installed as extension in the Python-3.6.4 easyconfig. can you confirm this in your site?

the reason for that (I think) is that setuptools-38.4.0 should have been installed with pip in order to override the setuptools that comes with Python itself. however, changing the base Python installation may be a bit drastic.

a more pragmatic solution is to just add setuptools as extension to hypothesis (this worked for me):
```
    ('setuptools', '38.4.0', {
        'source_tmpl': '%(name)s-%(version)s.zip',
        'source_urls': ['https://pypi.python.org/packages/source/s/setuptools/'],
        'checksums': ['6501fc32f505ec5b3ed36ec65ba48f1b975f52cf2ea101c7b73a08583fd12f75'],
    }),
```

**Comment by smoors** [comment]
@boegelbot please test @ generoso

**Comment by SimonPinches** [comment]
@smoors I've added your installation fix using `setuptools` and also bumped the version of hypothesis to 4.39.0 to get around some problems here.  This is the last version that works with `pytest/3.8.0` and doesn't require `pytest/4.3.0`.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3404.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/09c19bf07917456e9d876ae305cca447 for a full test report.

**Comment by boegel** [comment]
@smoors Do you mind following up on this?

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node376.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/5ea66e87e1a369d165de8ffcebc219e3 for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @SimonPinches!

**Pull request merged**